src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -582,12 +506,10 @@ export function syncModsFromEquipped(store: DimStore): LoadoutUpdateFunction {
     (item) => item.equipped && itemCanBeInLoadout(item) && item.bucket.sort === 'Armor'
   );
   for (const item of equippedArmor) {
-    mods.push(...extractArmorModHashes(item).map(mapToNonReducedModCostVariant));
```

Comment:
Switching to calling `updateMods` means this mapping only has to be done in one place.